### PR TITLE
DebugInfo generics inlining fix

### PR DESCRIPTION
**Pull request by slavapestov**
The SILOptimizer part is unrelated and hasn't caused any problems yet, but something I noticed by inspection.

rdar://problem/30786168

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@adrian-prantl and @swiftix, please take a look.

File: lib/IRGen/DebugTypeInfo.cpp [hunk 1]
```diff
@@ -50,16 +50,15 @@ DebugTypeInfo DebugTypeInfo::getFromTypeInfo(DeclContext *DC, swift::Type Ty,
...
                                               bool Unwrap) {
 
-  auto DeclType = Ty;
-  if (DeclCtx)
-    DeclType = (Decl->hasType()
-                    ? Decl->getType()
-                    : DeclCtx->mapTypeIntoContext(Decl->getInterfaceType()));
+  auto DeclType = (Decl->hasType()
+                   ? Decl->getType()
+                   : Decl->getDeclContext()->mapTypeIntoContext(
```
Comment (adrian-prantl) [inline lib/IRGen/DebugTypeInfo.cpp]: This is safe, because if the DeclContext has changed (i.e., we inlined generics) we will use RealType below.

**Review by adrian-prantl — COMMENTED** [review]
Thanks for fixing this!
LGTM

**Pull request merged**